citadel/indico_citadel/plugin.py

```diff
@@ -0,0 +1,71 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2002 - 2021 CERN
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from wtforms.fields.core import StringField
+from wtforms.fields.html5 import URLField
+from wtforms.validators import URL, DataRequired
+
+from indico.core import signals
+from indico.core.plugins import PluginCategory
+from indico.web.forms.base import IndicoForm
+from indico.web.forms.fields import IndicoPasswordField, TextListField
+
+from indico_citadel import _
+from indico_citadel.backend import LiveSyncCitadelBackend
+from indico_citadel.blueprint import blueprint
+from indico_citadel.cli import cli
+from indico_livesync import LiveSyncPluginBase
+
+
+class CitadelSettingsForm(IndicoForm):
+    search_backend_url = URLField(_('Citadel URL'), [DataRequired(), URL(require_tld=False)],
+                                  description=_('The URL of the Citadel server'))
+    search_backend_token = IndicoPasswordField(_('Citadel API token'), [DataRequired()], toggle=True,
+                                               description=_('The authentication token to access Citadel'))
+    search_owner_role = StringField(_('Search owner role'), [DataRequired()],
+                                    description=_('The role set on every synced object. This allows the members '
+                                                  'with that role to perform CRUD operations on the backend and '
+                                                  'synced objects.'))
+    file_extensions = TextListField(_('File extensions'),
+                                    description=_('The subset of file extensions that will be selected for indexing'))
+
+
+class CitadelPlugin(LiveSyncPluginBase):
+    """Citadel
+
+    Provides the search/livesync integration with Citadel
+    """
+
+    category = PluginCategory.search
+    configurable = True
+    settings_form = CitadelSettingsForm
+    default_settings = {
+        'search_backend_url': '',
+        'search_backend_token': '',
+        'search_owner_role': '',
```

Comment:
Probably not super important right now, but is there any sensible default value we can set which would work for a "simple" setup of citadel? Or can it be omitted altogether (in that case we'd need to make the field optional)?

Reply:
After reading https://cern-search.docs.cern.ch/usage/permissions/ it sounds like in our case we should put one of our egroups there? Or is this only if we want to somehow access the search data directly? Because then it sounds almost like leaving it empty (and handling that case properly, ie not sending an empty string) would be the way to go in most (all?) cases...